File: FEB_COOK19/2.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std ;

int main() {

    long long int t ;

    cin >> t ;

    while ( t-- ) {

        long long int n, ans = 0 ;

        cin >> n;

        vector< long long int > v(n) ;

        for ( auto i = 0 ; i < n ; ++i ) 
            cin >> v[i] ;

        sort( v.begin(), v.end() );

        for ( auto i = 0 ; i < n ; ++i ) {

            if ( ans >= v[i] )
                ans ++ ;
            else 
                break ; 

        }

        cout << ans << endl ;

        
    }

}

File: LTIME_NOV19B/2.py
def main() :
    t = int(input())
    ans = []

    while(t) :
        t -= 1
        n = int(input())
        l = [ int(i) for i in input().split() ]
        l.sort() 
        ol = len(l) ;
        s = ol ;
        i = 0
        while ( i < s ) :
            if ( i != 0 ) :
                if ( l[i]-1 != l[i-1] ) :
                    if ( i==s-1 or l[i+1] != l[i]+1 ) :
                        l.insert(i+1,l[i]+1)
                        s += 1
            elif ( l[1] != l[0]+1 ) :
                l.insert(1,l[0]+1)
                s += 1
                
            i += 1
        ans.append(s-ol) ;
        print(l)

    for i in ans :
        print(i)
            
main()    
    
